File: py/tree_node.py
class TreeNode:
  def __init__(self, path: str, size: int) -> None:
    self.children: list[TreeNode] = []
    self.path: str = path
    self.__size: int = size

  def append_child(self, childNode) -> None:
    self.children.append(childNode)

File: ts/fileSystem/TreeNode.ts
// Store name, parent, size and type in future?

export class TreeNode {
  #children: TreeNode[] = [];
  #path: string;

  constructor(path: string) {
    this.#path = path;
  };

  appendChild(childNode: TreeNode): void {
    this.#children.push(childNode);
  };

  deleteNode(): void {
    throw new Error('not implemented');
  };

  findPath(): void {
    throw new Error('not implemented');
  };

  generateTreeMap(): void {
    throw new Error('not implmented');
  };

  getAbsPath(): string {
    return this.#path;
  };

  printTree(): void {
    throw new Error('not implemented');
  };
};
